javaparser-core/src/main/java/com/github/javaparser/ast/body/CallableDeclaration.java

```diff
@@ -140,6 +148,15 @@ public T setParameters(final NodeList<Parameter> parameters) {
         return (T) this;
     }
 
+    public CallableDeclaration<T> setReceiverAnnotations(NodeList<AnnotationExpr> receiverAnnotations) {
+        this.receiverAnnotations = receiverAnnotations;
+        return this;
+    }
+
+    public NodeList<AnnotationExpr> getReceiverAnnotations() {
+        return receiverAnnotations;
```

Comment:
Is this something we should add to the core library?

Reply:
I think it would be good to have the receiver annotations list field. Because in case when receiver annotations passed to the method (for example) the javaparser keeps them at a regular parameter with name "this". The signature of the method is different when we pass the receiver annotations, but I think it should be the same.

Reply:
That's an interesting one. Made a suggestion in #1129

Reply:
Made a PR here: https://github.com/javaparser/javaparser/pull/1194